compiler/rustc_hir_analysis/src/check/wfcheck.rs

```diff
@@ -1104,6 +1104,23 @@ fn check_type_defn<'tcx>(
         for variant in variants.iter() {
             // All field types must be well-formed.
             for field in &variant.fields {
+                if let Some(def_id) = field.value
+                    && let Some(_ty) = tcx.type_of(def_id).no_bound_vars()
+                {
+                    if let Some(def_id) = def_id.as_local()
```

Comment:
leave a `FIXME(generic_const_exprs, default_field_values): ...` that says something about how this is a hack and to refactor this code to check the instantiate-ability of the code better.